### ISSUE TITLE
Regenerate assets on language changes

### ISSUE THREAD

**Issue by Cyberuben** [kirschbaum-development/laravel-translations-loader#21]
Hi,

is it possible to automatically trigger a webpack reload (I'm using --watch) when any of the PHP language files change?

**Comment by luisdalmolin** [comment]
Hi there, 

I think you'd need to include the location of the PHP files into the watcher maybe? I never really tried this, and not completely sure how it works.

I may look into this at some point, but since it's not a bug or anything I'll not put priority in this. If you find a solution let me know and I'll include in the docs.

**Comment by Cyberuben** [comment]
I can't find a single bit of documentation on how to properly add "watched" files or folders

**Comment by luisdalmolin** [comment]
Yeah, I tried to find something as well and didn't have much luck. In case I have more time at the end of the month I might take a look at this.

**Comment by RobertBoes** [comment]
I found quite an easy solution for this when you're using Laravel Mix

Install [extra-watch-webpack-plugin](https://github.com/pigcan/extra-watch-webpack-plugin). Then add the following to your `webpack.mix.js` file:
```js
const ExtraWatchWebpackPlugin = require('extra-watch-webpack-plugin');

mix.webpackConfig({
    plugins: [
        new ExtraWatchWebpackPlugin({
            dirs: [ 'resources/lang' ],
        }),
    ],
});
```
And now the watcher will run when files in the `resources/lang` directory change

**Comment by luisdalmolin** [comment]
Awesome, this looks good. I'm going to close this issue then. Thanks for posting back here.

**Comment by carlobeltrame** [comment]
Hey, sorry to revive a closed issue, but using a plugin that hasn't been updated in three years and does not support Webpack 5 does not seem a good solution to me.

Instead, I researched the correct way to register all files that should be watched with Webpack, and implemented them in a small [PR](#27). The [Webpack docs](https://webpack.js.org/api/loaders/#thisadddependency) even state that sass-loader and less-loader do this as well.
